merkle/src/merkle.rs

```diff
@@ -264,6 +522,54 @@ impl<T: Element, A: Algorithm<T>, K: Store<T>> MerkleTree<T, A, K> {
         }))
     }
 
+    /// Creates new merkle from an already allocated `Store` (used with
+    /// `DiskMmapStore::new_with_path` to set its path before instantiating
+    /// the MT, which would otherwise just call `DiskMmapStore::new`).
+    // FIXME: Taken from `MerkleTree::from_iter` to avoid adding more complexity,
+    //  it should receive a `parallel` flag to decide what to do.
+    // FIXME: We're repeating too much code here, `from_iter` (and
+    //  `from_par_iter`) should be extended to handled a pre-allocated `Store`.
+    pub fn from_data_with_store<I: IntoIterator<Item = T>>(
+        into: I,
+        data: &mut K,
```

Comment:
If the intent is to not clone, but consme the original this should be just `mut data: K`

Reply:
Thanks a lot, this was an unblocking comment.